File: python-sdk/src/epilot/models/shared/loginparameters.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from epilot import utils

class LoginParametersOauthResponseTypeEnum(str, Enum):
    CODE = "code"
    TOKEN = "token"


@dataclass_json
@dataclasses.dataclass
class LoginParameters:
    cognito_identity_pool_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cognito_identity_pool_id') }})
    cognito_oauth_domain: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cognito_oauth_domain') }})
    cognito_oauth_scopes: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cognito_oauth_scopes') }})
    cognito_region: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cognito_region') }})
    cognito_user_pool_client_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cognito_user_pool_client_id') }})
    cognito_user_pool_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cognito_user_pool_id') }})
    oauth_response_type: Optional[LoginParametersOauthResponseTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('oauth_response_type') }})
    organization_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('organization_id') }})
    organization_name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('organization_name') }})
    organization_type: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('organization_type') }})
    

File: user-api/java-sdk/lib/src/main/java/models/shared/LoginParameters.java
package .models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
public class LoginParameters {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("cognito_identity_pool_id")
    public String cognitoIdentityPoolId;
    public LoginParameters withCognitoIdentityPoolId(String cognitoIdentityPoolId) {
        this.cognitoIdentityPoolId = cognitoIdentityPoolId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("cognito_oauth_domain")
    public String cognitoOauthDomain;
    public LoginParameters withCognitoOauthDomain(String cognitoOauthDomain) {
        this.cognitoOauthDomain = cognitoOauthDomain;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("cognito_oauth_scopes")
    public String[] cognitoOauthScopes;
    public LoginParameters withCognitoOauthScopes(String[] cognitoOauthScopes) {
        this.cognitoOauthScopes = cognitoOauthScopes;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("cognito_region")
    public String cognitoRegion;
    public LoginParameters withCognitoRegion(String cognitoRegion) {
        this.cognitoRegion = cognitoRegion;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("cognito_user_pool_client_id")
    public String cognitoUserPoolClientId;
    public LoginParameters withCognitoUserPoolClientId(String cognitoUserPoolClientId) {
        this.cognitoUserPoolClientId = cognitoUserPoolClientId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("cognito_user_pool_id")
    public String cognitoUserPoolId;
    public LoginParameters withCognitoUserPoolId(String cognitoUserPoolId) {
        this.cognitoUserPoolId = cognitoUserPoolId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("oauth_response_type")
    public LoginParametersOauthResponseTypeEnum oauthResponseType;
    public LoginParameters withOauthResponseType(LoginParametersOauthResponseTypeEnum oauthResponseType) {
        this.oauthResponseType = oauthResponseType;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("organization_id")
    public String organizationId;
    public LoginParameters withOrganizationId(String organizationId) {
        this.organizationId = organizationId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("organization_name")
    public String organizationName;
    public LoginParameters withOrganizationName(String organizationName) {
        this.organizationName = organizationName;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("organization_type")
    public String organizationType;
    public LoginParameters withOrganizationType(String organizationType) {
        this.organizationType = organizationType;
        return this;
    }
}